channeldb/migration14/migration.go

```diff
@@ -0,0 +1,90 @@
+package migration14
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/coreos/bbolt"
+)
+
+// MigrateCommitDiff migrates the CommitDiff struct on-disk for all active
+// channels to store the CommitSig with a length prefix, just as we do on the
+// wire. This ensures we'll be able to store any extra data appended to the
+// message, yet still serialize items after the message as well on disk.
+func MigrateCommitDiff(tx *bbolt.Tx) error {
+	openChanBucket := tx.Bucket(openChannelBucket)
+	if openChanBucket == nil {
+		return nil
+	}
+
+	return openChanBucket.ForEach(func(nodePub, v []byte) error {
+		// Ensure that this is a key the same size as a pubkey,
+		// and also that it leads directly to a bucket.
+		if len(nodePub) != 33 || v != nil {
+			return nil
+		}
+
+		nodeChanBucket := openChanBucket.Bucket(nodePub)
+		if nodeChanBucket == nil {
+			return nil
+		}
+
+		// The next layer down is all the chains that this node
+		// has channels on with us.
+		return nodeChanBucket.ForEach(func(chainHash, v []byte) error {
+			// If there's a value, it's not a bucket so
+			// ignore it.
+			if v != nil {
+				return nil
+			}
+
+			chainBucket := nodeChanBucket.Bucket(chainHash)
+			if chainBucket == nil {
+				return fmt.Errorf("unable to read "+
+					"bucket for chain=%x", chainHash[:])
+			}
+
+			return chainBucket.ForEach(func(chanPoint, v []byte) error {
+				// If there's a value, it's not a bucket so
+				// ignore it.
+				if v != nil {
+					return nil
+				}
+
+				// Once we've found a valid channel bucket,
+				// we'll extract it from the node's chain
+				// bucket.
+				chanBucket := chainBucket.Bucket(chanPoint)
```

Comment:
missing nil check?

Reply:
If this is nil, then the user has issues with their database, as there's an invalid key at this level. Added the nil check nevertheless.